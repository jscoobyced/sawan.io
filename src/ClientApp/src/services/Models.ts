import { Url } from "url";

export enum Language {
    English,
    French
}

export interface Link {
    text: string;
    url: string;
    target?: string;
    title?: string;
}

export interface AllContent {
    navigationMenuContent: NavigationMenuContent;
    homeContent: HomeContent;
    menuContent: MenuContent;
}

export interface NavigationMenuContent {
    home: string | string[];
    websiteName: string | string[];
    applications: string | string[];
    cryptoCurrency: string | string[];
    healthMonitor: string | string[];
    about: string | string[];
}

export interface HomeContent {
    articleTitle: string | string[];
    article: string | string[];
}

export interface MenuContent {
    links: Link[];
}