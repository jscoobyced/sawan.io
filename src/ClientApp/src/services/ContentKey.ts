export class ContentKey {

    // General
    public static WebsiteName = 'WEBSITENAME';

    // Navigation Menu
    public static Home = 'HOME';
    public static About = 'ABOUT';
    public static Applications = 'APPLICATIONS';
    public static CryptoCurrency = 'CRYPTOCURRENCY';
    public static HealthMonitor = 'HEALTHMONITOR';

    // Home
    public static MainArticleTitle = 'MAINARTICLETITLE';
    public static MainArticle = 'MAINARTICLE';

    // Footer
    public static FooterCopyright = 'FOOTERCOPYRIGHT';
    public static FooterCredits = 'FOOTERCREDITS';
    public static FooterYear = 'FOOTERYEAR';
}
