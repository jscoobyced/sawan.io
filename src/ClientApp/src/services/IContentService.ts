import { BlogPage, Language, MainContent } from './Models';

export interface IContentService {
    getMainContent(language: Language): Promise<MainContent>;
    getDefaultMainContent(): MainContent;
    getBlogPage(maxResult: number, from: number): Promise<BlogPage>;
    getDefaultBlogPage(): BlogPage;
}
