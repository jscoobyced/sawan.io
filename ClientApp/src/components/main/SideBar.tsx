import React from 'react';
import { MainContent } from '../../services/Models';

export interface SideBarProps {
    allContent: MainContent;
}

export class SideBar extends React.Component<SideBarProps> {

    constructor(props: SideBarProps) {
        super(props);
    }

    public render() {
        const contentList = this.props.allContent.menuContent.links.map((link, key) => <li key={key}>
            <a href={link.url} target={link.target} title={link.title} >{link.text}</a>
        </li>);
        const linkContent = <ul>
            {contentList}
        </ul>;

        return <aside>
            {linkContent}
            <article>
                <h1>About C&eacute;dric</h1>
                <div>
                    I am a passionated software engineer that enjoys web development, web security,
                    Android development... and a bunch of other stuff.
                    <br />I live in The Land Of Smile (Thailand) and currently work
                    at <a href='https://www.agoda.com' target='_blank'>Agoda</a> as a Software Development Manager.
                </div>
            </article>
            <ul>
                <li>Source Code hosted in <a href='https://github.com/jscoobyced/sawan.io' target='_blank'>
                    <img src='github.png' title='Source code hosted in GitHub' alt='GitHub' />
                </a>
                </li>
                <li>Website built with <a href='https://ci.appveyor.com/project/jscoobyced/sawan-io' target='_blank'>
                    <img src='appveyor.png' title='Webiste built with AppVeyor' alt='AppVeyor' />
                </a>
                </li>
                <li>Code quality by <a href='https://codecov.io/gh/jscoobyced/sawan.io' target='_blank'>
                    <img src='codecov.png' title='Code quality with CodeCov' alt='CodeCov' />
                </a> and <a href='https://sonarcloud.io/dashboard?id=sawan' target='_blank'>
                        <img src='sonarqube.png' title='Code quality with SonarQube' alt='SonarQube' />
                    </a>
                </li>
            </ul>
        </aside>;
    }
}
