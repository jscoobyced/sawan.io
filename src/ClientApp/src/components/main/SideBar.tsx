import React from 'react';
import { Link, MenuContent } from '../../services/Models';

export interface SideBarProps {
    menuContent?: MenuContent;
}

export class SideBar extends React.Component<SideBarProps> {

    constructor(props: SideBarProps) {
        super(props);
    }

    public render() {
        const links: Link[] = this.props.menuContent ? this.props.menuContent.links : [];
        let linkContent;
        if (links) {
            const contentList = links.map((link, key) => <li key={key}>
                <a href={link.url} target={link.target} title={link.title} >{link.text}</a>
            </li>);
            linkContent = <ul>
                {contentList}
            </ul>;
        }

        return <aside>
            {linkContent}
            <article>
                <h1>About C&eacute;dric</h1>
                <div>
                    I am a passionated software engineer that enjoys web development, web security, 
                    Android development... and a bunch of other stuff.
                    <br />I live in The Land Of Smile (Thailand) and currently work 
                    at <a href='https://www.agoda.com' target='_blank'>Agoda</a> as a Software Development Manager.
                </div>
            </article>
            <ul>
                <li>Source Code hosted in <a href='https://github.com/jscoobyced/sawan.io' target='_blank'>
                    <img src='github.png' title='Source code hosted in GitHub' alt='GitHub' />
                </a>
                </li>
                <li>Website built with <a href='https://ci.appveyor.com/project/jscoobyced/sawan-io' target='_blank'>
                    <img src='appveyor.png' title='Webiste built with AppVeyor' alt='AppVeyor' />
                </a>
                </li>
                <li>Code quality by <a href='https://codecov.io/gh/jscoobyced/sawan.io' target='_blank'>
                    <img src='codecov.png' title='Code quality with CodeCov' alt='CodeCov' />
                </a> and <a href='https://sonarcloud.io/dashboard?id=sawan' target='_blank'>
                        <img src='sonarqube.png' title='Code quality with SonarQube' alt='SonarQube' />
                    </a>
                </li>
            </ul>
        </aside>;
    }
}