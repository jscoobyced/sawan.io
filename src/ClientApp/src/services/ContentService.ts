import { ApiUtils } from '../utils/ApiUtils';
import { IContentService } from './IContentService';
import { BlogElement, BlogPage, Language, MainContent } from './Models';

export class ContentService implements IContentService {

    public async getMainContent(language: Language): Promise<MainContent> {
        return await this.updateMainContent(language);
    }

    public getDefaultMainContent(): MainContent {
        return this.defaultMainContent();
    }

    public async getBlogPage(from: number, maxResult: number): Promise<BlogPage> {
        return await this.updateBlogPage(from, maxResult);
    }

    public getDefaultBlogPage(): BlogPage {
        return {
            articles: []
        };
    }

    protected async updateMainContent(language: Language): Promise<MainContent> {
        return await ApiUtils.fetchData<MainContent>(`/api/Data/main/${language}`);
    }

    protected defaultMainContent(): MainContent {
        return {
            menuContent: {
                links: []
            },
            navigationMenuContent: {
                about: '',
                applications: '',
                cryptoCurrency: '',
                healthMonitor: '',
                home: '',
                websiteName: ''
            },
            footerContent: {
                copyright: '',
                credits: '',
                year: ''
            }
        };
    }

    protected async updateBlogPage(from: number, maxResult: number): Promise<BlogPage> {
        const empty = {
            articles: []
        };
        if (maxResult <= 0) {
            return empty;
        }
        const data = await ApiUtils.fetchData<BlogElement[]>(`/api/Data/blog/${from}/${maxResult}`);
        if (!data || data.length === 0) {
            return empty;
        }
        return {
            articles: data.slice(0, Math.min(maxResult, data.length))
        };
    }
}
