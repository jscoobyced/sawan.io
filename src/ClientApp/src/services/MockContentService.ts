import { ContentService } from './ContentService';
import { BlogElement, BlogPage, Language, MainContent } from './Models';

export class MockContentService extends ContentService {

    private readonly data = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do'
        + ' ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco'
        + ' aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit '
        + ' cillum dolore eu fugiat nulla pariatur.<br />Excepteur sint occaecat cupidatat non proident, '
        + ' culpa qui officia deserunt mollit anim id est laborum. Commodo qui imperdiet massa tincidunt.'
        + ' Non enim praesent elementum facilisis leo. Adipiscing diam donec adipiscing tristique. Velit'
        + ' euismod in pellentesque massa placerat. Interdum velit euismod in pellentesque. Phasellus'
        + ' vestibulum lorem sed risus ultricies tristique nulla. Egestas erat imperdiet sed euismod nisi'
        + ' porta lorem mollis aliquam. Nulla facilisi etiam dignissim diam quis enim lobortis scelerisque'
        + ' fermentum. Morbi tristique senectus et netus et malesuada fames.<br />Vitae suscipit tellus'
        + ' mauris a diam maecenas. Sed risus ultricies tristique nulla aliquet. Sagittis eu volutpat odio'
        + ' facilisis mauris sit amet. Tempor orci eu lobortis elementum nibh tellus molestie nunc non.'
        + ' Turpis nunc eget lorem dolor.';

    protected updateMainContent(language: Language): Promise<MainContent> {
        const content: MainContent = {
            menuContent: {
                links: []
            },
            navigationMenuContent: {
                about: 'About',
                applications: 'Applications',
                cryptoCurrency: 'Crypto-Currency',
                healthMonitor: 'Health Monitor',
                home: 'Home',
                websiteName: 'sawan.io'
            },
            footerContent: {
                copyright: 'Copyright',
                credits: 'sawan.io',
                year: '2018 - ' + new Date().getFullYear()
            }
        };
        return Promise.resolve(content);
    }

    protected updateBlogPage(maxResult: number): Promise<BlogPage> {
        const date = '2018-11-15';
        const displayDate = (new Date(date).toISOString().substring(0, 10));
        const articles: BlogElement[] = [
            {
                article: `Latest news - ${displayDate}`,
                articleTitle: 'Grand opening of sawan.io.<br />This website is about my'
                    + ' personal training on various web-technologies: TypeScript, webpack, ReactJS, Secure HTTP'
                    + '  headers... It includes a simple (and a bit naive) crypto-currency analyser: it can'
                    + ' suggest you to sell or buy. Use at your own risk, it is amateur predictions, you have'
                    + ' been warned.<br />There will be also an application to track health metrics (weight,'
                    + ' % body fat, vascular fat ratio...). Or maybe not, this might be my next project.<br />'
                    + ' This project is open-sourced on <atarget=\'_blank\''
                    + ' href=\'https://github.com/jscoobyced/sawan.io\'>Github</a>. Enjoy and have fun.'
            },
            {
                article: 'Lorem Ipsum',
                articleTitle: this.data
            },
            {
                article: 'Some other news',
                articleTitle: this.data
            }
        ];
        return Promise.resolve({ articles });
    }
}
