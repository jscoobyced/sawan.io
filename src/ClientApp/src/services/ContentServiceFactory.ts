import { ContentService } from './ContentService';
import { IContentService } from './IContentService';
import { MockContentService } from './MockContentService';

export class ContentServiceFactory {

    public static GetContentService(mode: string): IContentService {
        if (mode === 'production') {
            return new ContentService();
        }

        return new MockContentService();
    }
}