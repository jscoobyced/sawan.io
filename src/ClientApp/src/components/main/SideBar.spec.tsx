import { shallow } from 'enzyme';
import React from 'react';
import { ContentService } from '../../services/ContentService';
import { SideBar } from './SideBar';

test('SideBar base component is unchanged.', () => {
    const sideBar = shallow(<SideBar />);
    expect(sideBar).toMatchSnapshot();
});

test('Layout component with content is unchanged.', () => {
    const menuContent = new ContentService().defaultContent().menuContent;
    const sideBar = shallow(<SideBar menuContent={menuContent} />);
    expect(sideBar).toMatchSnapshot();
});
