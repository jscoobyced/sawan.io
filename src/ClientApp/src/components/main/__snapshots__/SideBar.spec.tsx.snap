// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component with content is unchanged. 1`] = `
<aside>
  <ul>
    <li
      key="0"
    >
      <a
        href=""
        target=""
        title=""
      />
    </li>
  </ul>
  <ul>
    <li>
      Source Code hosted in 
      <a
        href="https://github.com/jscoobyced/sawan.io"
        target="_blank"
      >
        <img
          alt="GitHub"
          src="github.png"
          title="Source code hosted in GitHub"
        />
      </a>
    </li>
    <li>
      Website built with 
      <a
        href="https://ci.appveyor.com/project/jscoobyced/sawan-io"
        target="_blank"
      >
        <img
          alt="AppVeyor"
          src="appveyor.png"
          title="Webiste built with AppVeyor"
        />
      </a>
    </li>
    <li>
      Code quality by 
      <a
        href="https://codecov.io/gh/jscoobyced/sawan.io"
        target="_blank"
      >
        <img
          alt="CodeCov"
          src="codecov.png"
          title="Code quality with CodeCov"
        />
      </a>
       and 
      <a
        href="https://sonarcloud.io/dashboard?id=sawan"
        target="_blank"
      >
        <img
          alt="SonarQube"
          src="sonarqube.png"
          title="Code quality with SonarQube"
        />
      </a>
    </li>
  </ul>
</aside>
`;

exports[`SideBar base component is unchanged. 1`] = `
<aside>
  <ul />
  <ul>
    <li>
      Source Code hosted in 
      <a
        href="https://github.com/jscoobyced/sawan.io"
        target="_blank"
      >
        <img
          alt="GitHub"
          src="github.png"
          title="Source code hosted in GitHub"
        />
      </a>
    </li>
    <li>
      Website built with 
      <a
        href="https://ci.appveyor.com/project/jscoobyced/sawan-io"
        target="_blank"
      >
        <img
          alt="AppVeyor"
          src="appveyor.png"
          title="Webiste built with AppVeyor"
        />
      </a>
    </li>
    <li>
      Code quality by 
      <a
        href="https://codecov.io/gh/jscoobyced/sawan.io"
        target="_blank"
      >
        <img
          alt="CodeCov"
          src="codecov.png"
          title="Code quality with CodeCov"
        />
      </a>
       and 
      <a
        href="https://sonarcloud.io/dashboard?id=sawan"
        target="_blank"
      >
        <img
          alt="SonarQube"
          src="sonarqube.png"
          title="Code quality with SonarQube"
        />
      </a>
    </li>
  </ul>
</aside>
`;
